Prototype Wireframes

sknep opened the pull request:
For other Fellows engs to review & offer feedback!

Commit:
reset to begin wireframing

Commit:
Add USWDS

Commit:
Make fellows styling a sibling to uswds

Commit:
update USWDS to 1.0

Commit:
add modified flex box grid instead of uswds grid, customize uswds styles

Commit:
Add style guide examples and USWDS defaults for previewing

Commit:
Clarify pages that work

Commit:
remove unneeded collection

sknep commented on _config.yml:
```diff
@@ -1,27 +1,37 @@
+  - README.md
+
+collections:
+  content:
```
I'm trying out one content folder, thinking it might be easier to set up content editing workflows if the content and data are in containing folders in the root:
* `_content`
* `_data`

sknep commented on _includes/__left-nav.html:
```diff
@@ -0,0 +1,30 @@
+      <a href="{{ site.baseurl }}/changelog/">{{ changelog.title }}</a>
+    </li>
+  </ul>
+</nav>
```
Not using left-nav right now, just keeping it for later

briaguya replied:
After this merge it'd be worth deleting the file and relying on git history to get it back when you need it.

sknep commented on _includes/page-layout--department-listing.html:
```diff
@@ -0,0 +1,24 @@
+<div class="col-md-8">
+{% for department_listing in site.data.departments  %}
```
This is where we look through all the yaml files in `_data/departments`.
Keeping each 'listing' a separate file might help with governance.

briaguya commented on _plugins/replace-regex.rb:
```diff
@@ -0,0 +1,10 @@
+module Jekyll
+	module RegexFilter
+		def replace_regex(input, regex_string, replace_string)
```
Just curious, where/how is this being used?

sknep replied:
I think it was in the cloudcannon theme that I started with on this branch, I can remove it.

Commit:
Updated Config and 13 other files

Commit:
Move to collections rather than _data

Commit:
Merge remote-tracking branch 'origin/wireframe-prototyping' into wireframe-prototyping

# Conflicts:
#	_config.yml

Commit:
Updated Config and 4 other files

Commit:
updated style guide

- removed sections and updated some of the text in html.md
- added buttons partial from uswds copy
- added variables partial
- updated typography styles and colors
- updated footer background color
- updated all.scss

Commit:
Updated Html

Commit:
update to barton color syntax

Commit:
Improve compatibility with cloudcannon

Commit:
Updated Config and 3 other files

Commit:
Updated '_departments/another-department.md' via CloudCannon

Commit:
Updated Another Department

Commit:
Fix richtext editing for cloudcannon

Commit:
Updated Another Department and 1 other file

Commit:
Test editable content regions

Commit:
Updated '_content/contact.md' via CloudCannon

Commit:
Updated Contact

Commit:
Test more in-context editing

Commit:
Updated '_layouts/default.html' via CloudCannon

Commit:
Updated '_layouts/default.html' via CloudCannon

Commit:
Ok don't use "editable" on front matter values

Commit:
Start icons

Commit:
Remove cloud cannon editables

Commit:
Updated Contact

Commit:
Add basic metadata

Commit:
Updated Config

Commit:
Updated '_departments/residential-plan-review.md' via CloudCannon

jcbpl reviewed (changes requested):
@sknep I'd like to give this a thorough look but it's a bit difficult with all the commits and especially with all the USWDS artifacts. Could you:

1. Separate the USWDS code in one of these ways:

    * Use the NPM package (see https://github.com/18F/web-design-standards#install-using-npm). IMO this is the best way, but one question it would be useful to clarify is whether CloudCannon's build process supports NPM packages. Even if it doesn't, though, that doesn't rule out using CloudCannon or NPM, it just means we might need to consider an alternate build process, like Deploybot or CI. The ACC project has an example of this.
    * Use the uswds-jekyll theme. I haven't used this but if you've basically just copied Zilker at this point then it seems like it may be functionally equivalent, and offer an officially supported way of including USWDS via gem.
    * As a last resort, just create a separate branch and add USWDS in its own commit, then rebase or copy your changes on to a branch off of that one, so they can be reviewed independently.

2. Disconnect or delete the Siteleaf site, so it stops committing YAML changes.

3. Clean up and combine the commit history to better communicate intent, per the dev guide.

I'm glad to get together in person or via hangout if you want to talk through anything!

patrickm02L committed:
Updated '_departments/another-department.md' via CloudCannon

Commit:
Updated Another Department

patrickm02L committed:
Updated '_departments/another-department.md' via CloudCannon

Commit:
Updated Another Department

patrickm02L committed:
Updated '_departments/another-department.md' via CloudCannon

Commit:
Updated Another Department

patrickm02L committed:
Updated '_departments/another-department.md' via CloudCannon

Commit:
Updated Another Department

Commit:
Add icons as svgs

SVG icons available as inline via include or external

Commit:
Updated Config and 8 other files

Commit:
Setup site header & localization

Commit:
Rough out project pages

Commit:
Start on glossary plugin

Commit:
Merge remote-tracking branch 'origin/wireframe-prototyping' into wireframe-prototyping

# Conflicts:
#	_config.yml

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

Commit:
Updated '_projects/shed.md' via CloudCannon

sknep commented:
See PR #14 for a much better diff :)
